ckanext/stadtzhtheme/dcat/profiles.py

```diff
@@ -203,9 +203,16 @@ def graph_from_dataset(self, dataset_dict, dataset_ref):
         # Themes
         groups = self._get_dataset_value(dataset_dict, 'groups')
         try:
-            group_id = groups[0].get('id')
-            theme_ids = self._themes(group_id)
-            for theme_id in theme_ids:
+            theme_ids = []
+            for group in groups:
+                group_id = group.get('id')
+                if self._themes(group_id)[0]:
+                    theme_ids.append(str(self._themes(group_id)[0]))
```

Comment:
You can write this whole block as a list comprehension and use itertools.chain.from_iterable to flatten the list:

```python
import itertools
theme_ids = set(itertools.chain.from_iterable([self._themes(group.get('id')) for group in 
groups]))
```